### PR TITLE
Jest: Use github reporter to get PR annotations for test failures

### PR DESCRIPTION
**Pull request by torkelo**
Removed the old JUNIT reporter as Drone does not have a test reporting feature. This adds another reporter that pushes new annotations to github checks API.

### CODE REVIEW

**Review by peterholmberg — APPROVED** [review]
Very nice!

File: public/app/features/dashboard/components/PanelEditor/utils.ts [hunk 1]
```diff
@@ -28,7 +28,7 @@ export function calculatePanelSize(mode: DisplayMode, width: number, height: num
 }
 
 export function supportsDataQuery(plugin: PanelPlugin | undefined): boolean {
-  return plugin?.meta.skipDataQuery === false;
+  return plugin?.meta.skipDataQuery === true;
```
Comment (mckn) [inline public/app/features/dashboard/components/PanelEditor/utils.ts:31]: Is this by accident?

**Review by mckn — APPROVED** [review]
LGTM - one small question but the `test:ci` looks good.

File: public/app/features/dashboard/components/PanelEditor/utils.ts (hunk 1 above)
Comment (torkelo) [reply public/app/features/dashboard/components/PanelEditor/utils.ts:31]: no, wanted to trigger a test failure

**Comment by torkelo** [comment]
Sadly the way drone executes PRs by merging them into master first makes this not work

**Pull request closed without merging**